optical/converter/converter.py

```diff
@@ -300,6 +300,103 @@ def convert_coco(
+        "job-name": job_name,
+        "class-map": id_to_class_map,
+        "creation-date": str(datetime.now()),
+        "type": "groundtruth/object-detection",
+    }
+    annotations = []
+    for idx, row_info in grouped_info.iterrows():
+        annotations.append(
+            {
+                "class_id": int(row_info["class_id"]),
+                "width": int(row_info["width"]),
+                "height": int(row_info["height"]),
+                "top": int(row_info["y_min"]),
+                "left": int(row_info["x_min"]),
+            }
+        )
+    # append annotations
+    manifest_dic[f"{job_name}"]["annotations"] = annotations
+
+    return manifest_dic
+
+
+def convert_sagemaker(
+    df: pd.DataFrame,
+    root: Union[str, os.PathLike, PosixPath],
+    has_image_split: bool = False,
+    copy_images: bool = False,
+    save_under: str = "annotations",
+    output_dir: Optional[Union[str, os.PathLike, PosixPath]] = None,
+    job_name: str = "optical",
+):
+    """converts to sagemaker from master dataframe
+
+    Args:
+        df (pd.DataFrame): the master df
+        root (Union[str, os.PathLike, PosixPath]): root directory of the source format
+        has_image_split (bool, optional): If the images are arranged under the splits. Defaults to False.
+        copy_images (bool, optional): Whether to copy the images to a different directory. Defaults to False.
+        save_under (str, optional): Name of the folder to save the target annotations. Defaults to "labels".
+        output_dir (Optional[Union[str, os.PathLike, PosixPath]], optional): Output directory for the target
+            annotation. Defaults to ``None``.
+        job_name(Optional[str]): manifest job name for the output file. Defaults to optical
+    """
+
+    output_dir = ifnone(output_dir, root, Path)
+    output_labeldir = output_dir / f"{save_under}"
+    output_imagedir = output_dir / "images"
+    output_labeldir.mkdir(parents=True, exist_ok=True)
+
+    splits = df.split.unique().tolist()
+    for split in splits:
+        output_subdir = output_labeldir / split
+        output_subdir.mkdir(parents=True, exist_ok=True)
+
+        split_df = df.query("split == @split").copy()
+
+        # drop images missing width or height information
+        hw_missing = split_df[pd.isnull(split_df["image_width"]) | pd.isnull(split_df["image_height"])]
+        if len(hw_missing) > 0:
+            warnings.warn(
+                f"{hw_missing['image_id'].nunique()} has height/width information missing in split `{split}`. "
+                + f"{len(hw_missing)} annotations will be removed."
+            )
+
+        split_df = split_df[pd.notnull(split_df["image_width"]) & pd.notnull(split_df["image_height"])]
+
+        id_to_class_map = get_id_to_class_map(split_df)
+        grouped_split_df = split_df.groupby(["image_id", "width", "height"])
+
+        with open(os.path.join(output_subdir, f"{split}.manifest"), "w") as f:
```

Comment:
Please use `pathlib` instead of `os` for consistency with the rest of the module.